python/GangaLHCb/__init__.py

```diff
@@ -81,17 +62,26 @@ def _store_root_version():
 
 
 def _store_dirac_environment():
-    from GangaDirac.Lib.Utilities.DiracUtilities import write_env_cache, get_env
-    diracversion = _guess_version('LHCBDIRAC')
+    from GangaDirac.Lib.Utilities.DiracUtilities import write_env_cache
     platform = os.environ['CMTOPT']
+    diracversion = Ganga.Utility.Config.getConfig('LHCb')['LHCbDiracVersion']
     fdir = os.path.join(os.path.expanduser("~/.cache/Ganga/GangaLHCb"), platform)
     fname = os.path.join(fdir, diracversion)
     if not os.path.exists(fname) or not os.path.getsize(fname):
-        logger.info("Storing new LHCbDirac environment (%s:%s)" % (str(diracversion), str(platform)))
-        cmd = 'lb-run LHCBDIRAC {version} python -c "import os; print(dict(os.environ))"'.format(version=diracversion)
-        env = execute(cmd)  # grab the stdout text
-        env = eval(env)  # env is a string so convert it to a dict
+      cmd =  'lb-run LHCBDIRAC {version} python -c "import os; print(dict(os.environ))"'.format(version=diracversion)
+      env = execute(cmd)
+      try:
+        env = eval(env)
+      except SyntaxError:
+        logger.error("LHCbDirac version {version} does not exist".format(version=diracversion))
+        raise OptionValueError()
```

Comment:
This should probably be raised with some message like:
```python
raise OptionValueError('LHCbDirac version {version} does not exist'.format(version=diracversion))
```
so that when it's caught it can be displayed usefully.